### PR TITLE
CASSANDRA-16302 Fix flaky to reflect Array backed clustering

### PR DESCRIPTION
**Pull request by bereng**

### CODE REVIEW

**Comment by bereng** [comment]
At some point we switched to Array backed clusterings while the test was expecting ByteBuffers.

**Comment by bereng** [comment]
CI [j11](https://app.circleci.com/pipelines/github/bereng/cassandra/185/workflows/a9caeda6-e418-4207-9f7d-da1687a697c2)
CI [j8](https://app.circleci.com/pipelines/github/bereng/cassandra/185/workflows/de503497-b4cb-47b7-943c-9f4c97011a71)

The test that have been fixed only run on utetsts_long phase but including both full CI runs for completion.

File: src/java/org/apache/cassandra/db/commitlog/AbstractCommitLogSegmentManager.java [hunk 1]
```diff
@@ -506,7 +506,8 @@ public void awaitTermination() throws InterruptedException
         for (CommitLogSegment segment : activeSegments)
             segment.close();
 
-        bufferPool.emptyBufferPool();
+        if (bufferPool != null)
+            bufferPool.emptyBufferPool();
```
Comment (blerer) [inline src/java/org/apache/cassandra/db/commitlog/AbstractCommitLogSegmentManager.java:510]: I have not checked if `org.apache.cassandra.streaming.LongStreamingTest` also fail in 3.11 but we should probably backport that fix anyway.

File: test/long/org/apache/cassandra/hints/HintsWriteThenReadTest.java [hunk 2]
```diff
@@ -129,10 +130,10 @@ private void verifyHints(File directory, HintsDescriptor descriptor)
 
                     Row row = mutation.getPartitionUpdates().iterator().next().iterator().next();
                     assertEquals(1, Iterables.size(row.cells()));
-                    assertEquals(bytes(index), row.clustering().get(0));
+                    assertTrue(Arrays.equals(bytes(index).array(), (byte[]) row.clustering().get(0)));
```
Comment (blerer) [inline test/long/org/apache/cassandra/hints/HintsWriteThenReadTest.java]: This test was broken by the changes in CASSANDRA-15393 which introduced the `ArrayClustering` type .
If for some reason we switch to another Clustering that use ByteBuffers the test will fail again.
Should we not use  something like:

```
assertEquals(bytes(index), toByteBuffer(row.clustering().get(0)));
...
private ByteBuffer toByteBuffer(Object obj)
{
    if (obj instanceof ByteBuffer)
       return (ByteBuffer) obj;

    return ByteBuffer.wrap((byte[]) obj);
}
```

Comment (bereng) [reply test/long/org/apache/cassandra/hints/HintsWriteThenReadTest.java]: sgtm +1

File: src/java/org/apache/cassandra/db/commitlog/AbstractCommitLogSegmentManager.java (hunk 1 above)
Comment (bereng) [reply src/java/org/apache/cassandra/db/commitlog/AbstractCommitLogSegmentManager.java:510]: Doesn't fail there, but I can indeed add it.

**Comment by bereng** [comment]
CI [j11](https://app.circleci.com/pipelines/github/bereng/cassandra/186/workflows/a9fadaae-b6bc-4ffb-8220-fadf91cf52dd)
CI [j8](https://app.circleci.com/pipelines/github/bereng/cassandra/186/workflows/bb13466d-cee2-4f86-8783-891e4e4a2d90)

**Review by blerer — APPROVED** [review]

**Comment by blerer** [comment]
Committed manually.

**Pull request closed without merging**